src/input/label.component.ts

```diff
@@ -16,77 +16,90 @@ import { TextArea } from "./text-area.directive";
+			</ng-template>
 		</div>
-		<div
-			*ngIf="!skeleton && helperText && !invalid && !warn"
-			class="bx--form__helper-text"
-			[ngClass]="{'bx--form__helper-text--disabled': disabled}">
-			<ng-container *ngIf="!isTemplate(helperText)">{{helperText}}</ng-container>
-			<ng-template *ngIf="isTemplate(helperText)" [ngTemplateOutlet]="helperText"></ng-template>
-		</div>
-		<div *ngIf="!warn && invalid" class="bx--form-requirement">
-			<ng-container *ngIf="!isTemplate(invalidText)">{{invalidText}}</ng-container>
-			<ng-template *ngIf="isTemplate(invalidText)" [ngTemplateOutlet]="invalidText"></ng-template>
-		</div>
-		<div *ngIf="!invalid && warn" class="bx--form-requirement">
-			<ng-container *ngIf="!isTemplate(warnText)">{{warnText}}</ng-container>
-			<ng-template *ngIf="isTemplate(warnText)" [ngTemplateOutlet]="warnText"></ng-template>
+		<div class="bx--text-input__field-outer-wrapper"
+			[ngClass]="{
+			'bx--text-input__field-outer-wrapper--inline': inline
+			}">
+			<div
+				[class]="wrapperClass"
+				[ngClass]="{
+					'bx--text-input__field-wrapper--warning': warn
+				}"
+				[attr.data-invalid]="(invalid ? true : null)"
+				#wrapper>
+				<svg
+					*ngIf="!warn && invalid && !readonly"
+					ibmIcon="warning--filled"
+					size="16"
+					[ngClass]="{
+						'bx--text-input__invalid-icon': !textArea,
+						'bx--text-area__invalid-icon': textArea
+					}">
+				</svg>
+				<svg
+					*ngIf="!invalid && warn && !readonly"
+					ibmIcon="warning--alt--filled"
+					size="16"
+					class="bx--text-input__invalid-icon bx--text-input__invalid-icon--warning">
+				</svg>
+				<svg
+					*ngIf="readonly"
+					ibmIcon="edit--off"
+					size="16"
+					class="bx--text-input__readonly-icon">
+				</svg>
+				<ng-content select="input,textarea,div"></ng-content>
+			</div>
+			<ng-container *ngIf="!inline"
+				[ngTemplateOutlet]="helperTextTemplate" [ngTemplateOutletContext]="{inline: inline}">
+			</ng-container>
 		</div>
+
+		<ng-template #helperTextTemplate let-inline=inline>
+			<div *ngIf="!skeleton && helperText && ((!invalid && !warn) || inline)"
+				class="bx--form__helper-text"
+				[ngClass]="{
+					'bx--form__helper-text--disabled': disabled
+				}">
+				<ng-container *ngIf="!isTemplate(helperText)">{{helperText}}</ng-container>
+				<ng-template *ngIf="isTemplate(helperText)" [ngTemplateOutlet]="helperText"></ng-template>
+			</div>
+			<div *ngIf="!inline && !warn && invalid" class="bx--form-requirement">
+				<ng-container *ngIf="!isTemplate(invalidText)">{{invalidText}}</ng-container>
+				<ng-template *ngIf="isTemplate(invalidText)" [ngTemplateOutlet]="invalidText"></ng-template>
+			</div>
+			<div *ngIf="!inline && !invalid && warn" class="bx--form-requirement">
+				<ng-container *ngIf="!isTemplate(warnText)">{{warnText}}</ng-container>
+				<ng-template *ngIf="isTemplate(warnText)" [ngTemplateOutlet]="warnText"></ng-template>
+			</div>
+		<ng-template>
```

Comment:
I don't think this template takes `TextArea` into account. It's applying classes designed for TextInput to the entire label component. if the styles are changed on carbon's end, this would be a breaking change. Perhaps it would be best if you create a template for inline & swap to default if not? Thoughts?

Reply:
Just an FYI, Textarea does NOT support inline styling, It _WORKS_, but we shouldn't add a feature like this if it is not in Carbon guideline.

Reply:
Oh Ok, I actually wasn't sure whether textarea was supposed to have the inline styling or not... I asked it in the issue as well -- https://github.com/carbon-design-system/carbon-components-angular/issues/2291#issuecomment-1458418523
So, I'll remove the textarea part.

Reply:
Oh sorry! I didn't see the notification for that issue.